### DISCUSSION TITLE
Search page redirections

### DISCUSSION THREAD

**Discussion by obahchimaobi** [livewire/livewire#9087] [1. Help]
How do I redirect to a search page when a user clicks the enter button using livewire?. I already made a real-time search.

**Comment by danie-ramdhani** [comment]
something like this?

```blade
<button type="button" @keyup.enter.window="$wire.redirectToSearchPage">
    Redirect
</button>
```

```php
public function redirectToSearchPage()
{
    $this->redirect('/search-page');
}
```

**Reply by obahchimaobi** [reply]
There's no button. It's just an input form like this:

```
<input wire:model.debounce='searchBar'/>
```
Now as the user types a search keyword, it displays the results. That's the real time search.

Because I am using livewire component to handle displaying of results when the user hits enter, the input query becomes empty thereby resulting to displaying all items in my DB.
I used dump() and saw it.

**Comment by obahchimaobi** [comment]
I'll show you my code for you to review it

**Comment by EdgarsJoja** [comment]
Wrap the input in form, then you can add `wire:submit` or `@submit` handlers. Do the redirect in handlers.

**Reply by obahchimaobi** [reply]
Okay.
Let me do it now.
I'll notify you if i come across any other challenges

**Reply by obahchimaobi** [reply]
Here is my code:

`
use WithoutUrlPagination, WithPagination;

    public $searchBar = '';

    public function search()
    {
        $movies = Movies::search($this->searchBar)->get();

        dump($this->searchBar);

        $series = Series::search($this->searchBar)->get();

        $allResults = $movies->merge($series);

        $page = LengthAwarePaginator::resolveCurrentPage() ?: 1;

        // Items per page
        $perPage = 24;

        // Slice the collection to get the items to display in current page
        $currentPageResults = $allResults->slice(($page * $perPage) - $perPage, $perPage)->values();

        // Create our paginator and add it to the view
        $paginatedResults = new LengthAwarePaginator($currentPageResults, count($allResults), $perPage, $page, ['path' => LengthAwarePaginator::resolveCurrentPath()]);

        return view('media.search', [
            'paginatedResults' => $paginatedResults,
        ]);
    }

    public function render()
    {
        // dump($this->search);
        $results = [];

        if (strlen($this->searchBar) >= 1) {
            $movies_results = Movies::where('name', 'like', '%'.$this->searchBar.'%')
                ->whereNull('deleted_at')
                ->where('status', '!=', 'pending')
                ->select(['name', 'formatted_name', 'poster_path', 'release_year']);

            $results = Series::where('name', 'like', '%'.$this->searchBar.'%')
                ->whereNull('deleted_at')
                ->where('status', '!=', 'pending')
                ->select(['name', 'formatted_name', 'poster_path', 'release_year'])->union($movies_results)->simplePaginate('20');
            // dump($results);
        }

        $error = '';

        if (count($results) == 0) {
            // code...
            $error = 'No result found';
        }

        return view('livewire.search-bar', compact('results', 'error'));
    }
`

**Comment by obahchimaobi** [comment]
I was finally able to achieve this. thanks so much guys
